<a href="https://colab.research.google.com/github/rahiakela/practical-machine-learning-with-tensorflow/blob/week-1/begining_tensorflow2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Begining with TensorFlow 2.0

Download and install the TensorFlow 2.0 RC package. Import TensorFlow into your program:

In [2]:
from __future__ import absolute_import, division, print_function, unicode_literals

# Install TensorFlow
!pip install tensorflow==2.0.0-beta1

In [3]:
import tensorflow as tf

# check tensorflow version
print(tf.__version__)

2.0.0-beta1


Load and prepare the MNIST dataset. Convert the samples from integers to floating-point numbers:

In [6]:
mnist = tf.keras.datasets.mnist

# prepare training and test set
(x_train, y_train),(x_test, y_test) = mnist.load_data()

# normalize training and test set
x_train, x_test = x_train / 255.0, x_test / 255.0

11493376/11490434 [==============================] - 0s 0us/step


Build the tf.keras.Sequential model by stacking layers. Choose an optimizer and loss function for training:

In [0]:
# create NN model architecture
model = tf.keras.models.Sequential([
    tf.keras.layers.Flatten(input_shape=(28, 28)),
    tf.keras.layers.Dense(128, activation='relu'),
    tf.keras.layers.Dropout(0.2),
    tf.keras.layers.Dense(10, activation='softmax')
])

# compile the model
model.compile(optimizer='adam', loss='sparse_categorical_crossentropy', metrics=['accuracy'])

Train and evaluate the model:

In [9]:
model.fit(x_train, y_train, epochs=5)  # passing training features and label matrix and iteration value
model.evaluate(x_test, y_test)         # passing testing features and label matrix

W0826 13:22:00.387919 140324093978496 deprecation.py:323] From /usr/local/lib/python3.6/dist-packages/tensorflow/python/ops/math_grad.py:1250: add_dispatch_support.<locals>.wrapper (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where


Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 6s 99us/sample - loss: 0.2989 - accuracy: 0.9130
Epoch 2/5
60000/60000 [==============================] - 6s 93us/sample - loss: 0.1438 - accuracy: 0.9566
Epoch 3/5
60000/60000 [==============================] - 6s 95us/sample - loss: 0.1069 - accuracy: 0.9674
Epoch 4/5
60000/60000 [==============================] - 6s 96us/sample - loss: 0.0863 - accuracy: 0.9736
Epoch 5/5
10000/10000 [==============================] - 0s 41us/sample - loss: 0.0725 - accuracy: 0.9765


[0.07247765306369401, 0.9765]

The image classifier is now trained to ~98% accuracy on this dataset.